source/appModules/explorer.py

```diff
@@ -195,6 +203,7 @@ def chooseNVDAObjectOverlayClasses(self, obj, clsList):
 			if role == controlTypes.ROLE_LIST:
 				clsList.remove(List)
 			clsList.insert(0, StartButton)
+			return

```

Comment:
Where's this one coming from?

Reply:
IIRC this short-circuits the function to avoid doing a bunch of checks that won't lead to any more code being executed.

Reply:
Can you add a comment here to explain this? When maintaining this code, it will be important to know that the early return here is purely an optimisation, and not functional. I guess something along the lines of: "Because the windowClass is start, and role is either list or button, none of the checks after this will apply, so we can return early"